src/main/java/org/takes/rs/BodyContent.java

```diff
@@ -0,0 +1,182 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Yegor Bugayenko
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.rs;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+
+/**
+ * The body of a response used by {@link RsWithBody}.
+ *
+ * @author Nicolas Filotto (nicolas.filotto@gmail.com)
+ * @version $Id$
+ * @since 0.31
+ */
+interface BodyContent {
+    /**
+     * Gives a reusable {@code InputStream} corresponding to the content of
+     * the body.
+     * @return A reusable {@code InputStream}.
+     * @throws IOException in case a reusable {@code InputStream} could not
+     *  be provided.
+     */
+    InputStream input() throws IOException;
+
+    /**
+     * Gives the length of the stream.
+     * @return The length of the stream.
+     * @throws IOException in case the length of the stream could not be
+     *  retrieved.
+     */
+    int length() throws IOException;
+
+    /**
+     * Content of a body based on an {@link URL}.
+     */
+    final class URLContent implements BodyContent {
+
+        /**
+         * The {@link URL} of the content.
+         */
+        private final transient URL url;
+
+        /**
+         * Constructs an {@code URLContent} with the specified {@link URL}.
+         * @param content The {@link URL} of the content.
+         */
+        URLContent(final URL content) {
+            this.url = content;
+        }
+
+        @Override
+        public InputStream input() throws IOException {
+            return this.url.openStream();
+        }
+
+        @Override
+        public int length() throws IOException {
+            try (final InputStream input = this.url.openStream()) {
+                return input.available();
```

Comment:
@krzyk Actually it is an existing code that I moved into this class as you can see [here](https://github.com/yegor256/takes/blob/31c1ad15ce4b2ed02b616e5f2fdce10033411ebe/src/main/java/org/takes/rs/RsWithBody.java#L120). The only thing that I added is the try-with-resources in order to close the stream which was not done before.